src/types/OfferItem.sol

```diff
@@ -1,9 +1,11 @@
 // SPDX-License-Identifier: MIT
 pragma solidity ^0.8.18;
 
-// @dev Struct representing an offer item (a token)
-// @dev We only support ERC721 on EVM chains for now
+// @dev Struct representing an offer item
+// @dev We support ERC721 and ERC20
+// @dev If amount is 0, then we use tokenId.
 struct OfferItem {
     address tokenAddress;
     uint256 tokenId;
+    uint256 amount;
```

Comment:
see my previous comment. I think I would simply add tokenType here and amount would always be 1 in the case of an ERC721

Reply:
or name the variable tokenIdOrAmount to save space...